src/osp/Universe.h

```diff
@@ -67,225 +67,89 @@ class Universe
-     * @param target [in]
-     * @return relative position of target in SpaceInt vector
+     * @brief Calculate position between two satellites.
+     *
+     * @param referenceFrame [in] Satellite used as reference frame
+     * @param target         [in] Satellite to calculate position to
+     * @return relative position of target in spaceint_t vector
      */
-    Vector3s sat_calc_pos(Satellite referenceFrame, Satellite target) const;
+    Vector3g sat_calc_pos(Satellite referenceFrame, Satellite target) const;
 
     /**
-     * Calculate position between two satellites.
-     * @param referenceFrame [in]
-     * @param target [in]
+     * @brief Calculate position between two satellites.
+     *
+     * @param referenceFrame [in] Satellite used as reference frame
+     * @param target         [in] Satellite to calculate position to
      * @return relative position of target in meters
      */
     Vector3 sat_calc_pos_meters(Satellite referenceFrame, Satellite target) const;
 
     /**
-     * Create a Trajectory, and add it to the universe.
-     * @tparam TRAJECTORY_T Type of Trajectory to construct
-     * @return Reference to new trajectory just created
+     * @brief Ressign indices in the UCompInCoordspace components of satellites
+     *        in a CoordinateSpace's m_toAdd queue
+     *
+     * @param coordSpace [in] Index to CoordinateSpace in m_coordSpaces
      */
-    template<typename TRAJECTORY_T, typename ... ARGS_T>
-    TRAJECTORY_T& trajectory_create(ARGS_T&& ... args);
+    void update_sat_coordspace(uint32_t coordSpace);
 
-    constexpr Registry_t& get_reg() noexcept { return m_registry; }
-    constexpr const Registry_t& get_reg() const noexcept
+    constexpr Reg_t& get_reg() noexcept { return m_registry; }
+    constexpr const Reg_t& get_reg() const noexcept
     { return m_registry; }
 
+    std::vector< std::optional<CoordinateSpace> > m_coordSpaces;
+
 private:
 
     Satellite m_root;
 
-    std::vector<std::unique_ptr<ISystemTrajectory>> m_trajectories;
-
-    Registry_t m_registry;
+    Reg_t m_registry;
 
 };
 
-template<typename TRAJECTORY_T, typename ... ARGS_T>
-TRAJECTORY_T& Universe::trajectory_create(ARGS_T&& ... args)
-{
-    std::unique_ptr<TRAJECTORY_T> newTraj
-                = std::make_unique<TRAJECTORY_T>(std::forward<ARGS_T>(args)...);
-    return static_cast<TRAJECTORY_T&>(*m_trajectories.emplace_back(std::move(newTraj)));
-}
 
 // default ECS components needed for the universe
 
-
 struct UCompTransformTraj
 {
-    // Relative to m_parent
-    Vector3s m_position;
-
-    // In 'global' space
-    Quaternion m_rotation;
-
-    Satellite m_parent; // Set only by trajectory
-
-    ISystemTrajectory *m_trajectory{nullptr}; // get rid of this some day
-    unsigned m_index; // index in trajectory's vector
-
-    // set this to true when you modify something the trajectory might use
-    bool m_dirty{true};
-
     // move this somewhere else eventually
     std::string m_name;
```

Comment:
Reason to keep UCompTransformTraj if all it holds is a name, especially one that has a comment saying it's going to be moved eventually?

Reply:
It's pretty much dead right now. Name should be separated into a UCompName, and rotation can be stored in the coordinate spaces.